specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/notebook.json

```diff
@@ -0,0 +1,384 @@
+					},
+					{
+						"$ref": "#/parameters/NotebookWorkspaceNameParameter"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "The operation completed successfully.",
+						"schema": {
+							"$ref": "#/definitions/NotebookWorkspaceConnectionInfoResult"
+						}
+					}
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/notebookWorkspaces/{notebookWorkspaceName}/regenerateAuthToken": {
+			"post": {
+				"tags": [
+					"NotebookWorkspacesResource"
+				],
+				"operationId": "NotebookWorkspaces_RegenerateAuthToken",
+				"x-ms-examples": {
+					"CosmosDBNotebookWorkspaceRegenerateAuthToken": {
+						"$ref": "./examples/CosmosDBNotebookWorkspaceRegenerateAuthToken.json"
+					}
+				},
+				"description": "Regenerates the auth token for the notebook workspace",
+				"parameters": [{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/subscriptionIdParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/resourceGroupNameParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/apiVersionParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/accountNameParameter"
+					},
+					{
+						"$ref": "#/parameters/NotebookWorkspaceNameParameter"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"202": {
+						"description": "Accepted. The regenerate auth token operation will complete asynchronously."
+					}
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/notebookWorkspaces/{notebookWorkspaceName}/start": {
+			"post": {
+				"tags": [
+					"NotebookWorkspacesResource"
+				],
+				"operationId": "NotebookWorkspaces_Start",
+				"x-ms-examples": {
+					"CosmosDBNotebookWorkspaceStart": {
+						"$ref": "./examples/CosmosDBNotebookWorkspaceStart.json"
+					}
+				},
+				"description": "Starts the notebook workspace",
+				"parameters": [{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/subscriptionIdParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/resourceGroupNameParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/apiVersionParameter"
+					},
+					{
+						"$ref": "../../stable/2019-08-01/cosmos-db.json#/parameters/accountNameParameter"
+					},
+					{
+						"$ref": "#/parameters/NotebookWorkspaceNameParameter"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"202": {
+						"description": "Accepted. The start operation will complete asynchronously."
+					}
+				}
+			}
+		}
+	},
+	"definitions": {
+		"NotebookWorkspaceCreateParameters": {
```

Comment:
This is for future-proofing. Overtime it is possible to have additional user provided parameters here.
Another reason we want to model this as a separate resource is that it represents a set of functionalities that are very distinct from those provided by the parent resource. We want to avoid bloating the parent resource too much. This resource also has its own set of actions that we want to keep separate from the parent to avoid confusion.